commcare_connect/commcarehq/api.py

```
@@ -0,0 +1,104 @@
+def update_case_data_by_case_id(api_key: HQApiKey, domain: str, case_id: str, data: dict[str, any]) -> dict[str, any]:
+    url = f"{api_key.hq_server.url}/a/{domain}/api/case/v2/{case_id}/"
+    response = httpx.put(
+        url,
+        headers={"Authorization": f"ApiKey {api_key.user.email}:{api_key.api_key}"},
+        json=data,
+    )
+
+    try:
+        response.raise_for_status()
+    except httpx.HTTPStatusError as e:
+        raise CommCareHQAPIException(f"Failed to update case data for {domain} with {case_id}. HQ Error: {e}")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)